### PR TITLE
light/txpool: removed redundant transaction existence check

### PR DESCRIPTION
**Pull request by joohhnnn**
I noticed that in the add function of txpool.go file in the Geth light client, there seems to be a redundant transaction existence check being performed. 
At first I thought the double-check could potentially be a strategy to handle concurrency issues that might occur during the validateTx function call. and I asked @jsvisa for help.

jsvisa's opinion: If a mutex is being used while accessing the pending map, concurrency should not be an issue, rendering the second check unnecessary. Additionally, he pointed out that the nonce map also encounters concurrent access and write operations, yet doesn't seem to have a separate lock mechanism in place.

Comparatively, we didn't observe a similar redundancy in the full client's txpool.go. As jsvisa pointed out, if this mutex has a broad impact, there should be no concurrency concerns during the validateTx process, making the second check seem unnecessary.

Adding to these, I've noticed that the logging mechanism after the second transaction check seems to be inconsistent. If the logic somehow skips over the `!ok` condition, thereby not adding the transaction to the pending map, the log message would still report the transaction as successfully added, which could lead to confusion.

In summary, we believe the transaction existence check might be redundant and the logging mechanism might need to be reconsidered. However, we also acknowledge that there could be design considerations unknown to us that necessitate this redundancy or logging logic.

We would appreciate any further insights or clarifications on this matter.

### CODE REVIEW

**Review by rjl493456442 — APPROVED** [review]
lgtm.

But I have to say, LES is not working since The Merge. We prefer to completely rewrite the entire protocol since it's a good timing. We may not accept small twists since it's not that useful. No offense.

**Comment by joohhnnn** [comment]
> lgtm.
> 
> But I have to say, LES is not working since The Merge. We prefer to completely rewrite the entire protocol since it's a good timing. We may not accept small twists since it's not that useful. No offense.

Could you provide a more detailed description of the entire task? I think Im glad to make some contributions to it.

**Comment by rjl493456442** [comment]
@zsfelfoldi can ask it. Now he is working on a new sync protocol based on proof-of-stake algorithm. Besides, we will consider to simplify some server side flow control mechanism and some other things.
